src/python/pants/option/global_options.py

```diff
@@ -44,7 +44,8 @@ def register_bootstrap_options(cls, register):
     register('-l', '--level', choices=['debug', 'info', 'warn'], default='info', recursive=True,
              help='Set the logging level.')
     register('-q', '--quiet', type=bool, recursive=True,
-             help='Squelches most console output.')
+             help='Squelches most console output. NOTE: Some tasks default to behaving quietly: '
+                  'inverting this option supports making them noisier than they would be otherwise.')
```

Comment:
Nice. Maybe mention here that --no-quiet is a good way to get more information in case of error?

Also, maybe a TODO somewhere to add to the help for each task whether it's quiet or not?

Reply:
I am leaving it as is, because simply putting it there without adequate explanation may cause confusion with `-ldebug` or `-lwarn`. 

Filed https://github.com/pantsbuild/pants/issues/4988 for todo.